apps/astarte_trigger_engine/lib/astarte_trigger_engine/events_consumer.ex

```diff
@@ -62,41 +62,65 @@ defmodule Astarte.TriggerEngine.EventsConsumer do
     handle_simple_event(realm, device_id, headers, event_type, event, timestamp_ms)
   end
 
-  defp handle_simple_event(realm, device_id, headers_map, event_type, event, timestamp_ms) do
+  defp handle_simple_event(
+         realm,
+         device_id,
+         headers_map,
+         event_type,
+         event,
+         timestamp_ms
+       ) do
     with {:ok, trigger_id} <- Map.fetch(headers_map, "x_astarte_parent_trigger_id"),
-         {:ok, action} <- retrieve_trigger_configuration(realm, trigger_id),
-         {:ok, payload} <-
-           event_to_payload(realm, device_id, event_type, event, action, timestamp_ms),
-         {:ok, headers} <- event_to_headers(realm, device_id, event_type, event, action),
-         :ok <- execute_action(payload, headers, action) do
-      :telemetry.execute(
-        [:astarte, :trigger_engine, :http_action_executed],
-        %{},
-        %{realm: realm, status: :ok}
-      )
+         {:ok, action} <- retrieve_trigger_configuration(realm, trigger_id) do
+      with {:ok, payload} <-
+             event_to_payload(realm, device_id, event_type, event, action, timestamp_ms),
+           {:ok, headers} <- event_to_headers(realm, device_id, event_type, event, action) do
+        payload_size = payload |> :erlang.term_to_binary() |> :erlang.byte_size()
+
+        case execute_action(payload, headers, action) do
+          :ok ->
+            :telemetry.execute(
+              [:astarte, :trigger_engine, :http_action_executed],
+              %{size: payload_size},
+              %{
+                realm: realm,
+                status: :ok
+              }
+            )
+
+            :ok
+
+          {:error, {:http_error, status_code}} ->
+            :telemetry.execute(
+              [:astarte, :trigger_engine, :http_action_executed],
+              %{size: payload_size},
+              %{realm: realm, status: status_code}
+            )
+
+            {:http_error, status_code}
+
+          {:error, :connection_error} ->
+            :telemetry.execute(
+              [:astarte, :trigger_engine, :http_action_executed],
+              %{size: payload_size},
+              %{realm: realm, status: :connection_error}
+            )
+
+            {:error, :connection_error}
+        end
```

Comment:
I'd suggest moving this in another function to decrease cognitive load when reading the code